tests/extensions/test_eo.py

```diff
@@ -83,6 +84,13 @@ def test_bands(self) -> None:
         self.assertEqual(len(EOExtension.ext(item).bands or []), 3)
         item.validate()
 
+    def test_asset_bands_s2(self) -> None:
+        item = pystac.Item.from_file(self.S2_ITEM_URI)
+        for key, asset in item.get_assets().items():
+            eo_bands = EOExtension.ext(asset).bands
+            if key == "mtd":
+                assert eo_bands is None
+
```

Comment:
Sorry I didn't catch this earlier. It seems like it would be simpler and faster to get the `"mtd"` asset directly rather than looping over all of the assets. Since we're not making any assertions on the bands for the other assets it doesn't seem necessary to construct the extension and get the `Bands` objects for those.

```suggestion
        item = pystac.Item.from_file(self.S2_ITEM_URI)
        mtd_asset = item.get_assets()["mtd"]
        self.assertIsNone(EOExtension.ext(asset).bands)
```

Reply:
yes. Initially, the test was intended to detect some "residual" objects from other assets